opencti-platform/opencti-graphql/src/utils/stix-filtering/boolean-logic-engine.ts

```diff
@@ -0,0 +1,187 @@
+      || (operator === 'gt' && adaptedFilterValues.some((v) => stixCandidates.some((c) => c > v)))
+      || (operator === 'gte' && adaptedFilterValues.some((v) => stixCandidates.some((c) => c >= v)));
+  }
+
+  return false;
+};
+
+/**
+ * Implementation of testGenericFilter for string values.
+ * String comparison is insensitive to case.
+ */
+export const testStringFilter = (filter: FilterExcerpt, stixCandidates: string[]) => {
+  const filterValuesLowerCase = filter.values.map((v) => v.toLowerCase());
+  const stixValuesLowerCase = stixCandidates.map((v) => v.toLowerCase());
+  return testGenericFilter<string>(filter, filterValuesLowerCase, stixValuesLowerCase);
+};
+
+/**
+ * Implementation of testGenericFilter for boolean values.
+ * Filter values are parsed as booleans
+ * The strings "true", "yes" or "1" are interpreted as true ; anything else is false.
+ */
+export const testBooleanFilter = (filter: FilterExcerpt, stixCandidate: boolean | null | undefined) => {
+  const filterValuesAsBooleans = filter.values.map((v) => v.toLowerCase() === 'true' || v.toLowerCase() === 'yes' || v === '1');
+  return testGenericFilter<boolean>(filter, filterValuesAsBooleans, toValidArray(stixCandidate));
+};
+
+/**
+ * Implementation of testGenericFilter for numerical values.
+ * Filter values are parsed as integers.
+ */
+export const testNumericFilter = (filter: FilterExcerpt, stixCandidate: number | null | undefined) => {
+  const filterValuesAsNumbers = filter.values.map((v) => parseInt(v, 10)).filter((n) => !Number.isNaN(n));
+  return testGenericFilter<number>(filter, filterValuesAsNumbers, toValidArray(stixCandidate));
+};
+
+/**
+ * Specific tester using the Moment.js library to compare dates.
+ */
+export const testDateFilter = ({ mode, operator, values }: FilterExcerpt, stixCandidate: string | null | undefined) => {
+  // make sure the dates are valid, otherwise we won't use them.
+  const filterValuesAsDates = values.map((v) => moment(new Date(v))).filter((d) => d.isValid());
+
+  if (operator === 'nil' || (operator === 'eq' && filterValuesAsDates.length === 0)) {
+    return stixCandidate === null;
+  }
+  if (operator === 'not_nil' || (operator === 'not_eq' && filterValuesAsDates.length === 0)) {
+    return stixCandidate !== null;
+  }
+
+  // excluding the cases above, comparing to nothing is not supported and would never match
+  if (stixCandidate === null || stixCandidate === undefined) {
+    return false;
+  }
+
+  const stixDate = moment(new Date(stixCandidate));
+  if (!stixDate.isValid()) {
+    // This is actually an error case that should not happen (invalid stix)
+    return false;
+  }
+
+  if (mode === 'and') {
+    // NOTE: equality is very strict (milliseconds)
+    return (operator === 'eq' && filterValuesAsDates.every((v) => stixDate.isSame(v)))
+      || (operator === 'not_eq' && filterValuesAsDates.every((v) => !stixDate.isSame(v)))
```

Comment:
isSame is maybe too strict as the filter selected in UI are only dates, no time.

Reply:
probably use isSameDay

Reply:
this has to be discussed a bit more : what's the expected granularity for date filtering. UI wise it's day, but API wise it could be millisec.

Reply:
seen with @richard-julien : our storage granularity is the millisec, our API when testing equality should be granular in millisecs too. We'll keep everything that way.

The UI shall be adapted to this fact, and not suggest to use the `eq / not_eq` operators when it comes to date (cc @jpkha).

As an evolution, let's think about a new operator `intersects`. 
`<created_at> intersects "2023-08-24"` would mean "creation date is in the **period** "24th August 2023".